src/fio/FioEngine.ts

```diff
@@ -198,6 +198,18 @@ export class FioEngine extends CurrencyEngine<FioTools, SafeFioWalletInfo> {
       fioAction: async (actionName: string, params: any): Promise<any> => {
         return await this.multicastServers(actionName, params)
       },
+      fetchFioAddresses: async (): Promise<FioAddress[]> => {
+        await this.refreshFioAddresses()
+        return this.otherData.fioAddresses
+      },
+      fetchFioAddressNames: async (): Promise<string[]> => {
+        await this.refreshFioAddresses()
+        return this.otherData.fioAddresses.map(fioAddress => fioAddress.name)
```

Comment:
This seems redundant with `fetchFioAddressNames`, since the client can just do the `map` itself. These methods are pretty heavy-weight now, since they query the network.

Reply:
I did consider going that route, that doesn't count as changing public APIs?